### PR TITLE
Add SecurityPriceInfo

### PR DESCRIPTION
**Pull request by jodastephen**
Permits the price of a security to be interpreted using tick size, tick value and contract size.

### CODE REVIEW

File: modules/product/src/main/java/com/opengamma/strata/product/SecurityPriceInfo.java [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+
+  /**
+   * The size of each tick.
+   * <p>
+   * Tick size is the minimum movement in the price of the security.
+   * For example, the price might move up or down in units of 0.01
+   * It must be a positive decimal number.
+   */
+  @PropertyDefinition(validate = "ArgChecker.notNegativeOrZero")
+  private final double tickSize;
```
Comment (cjkent) [inline modules/product/src/main/java/com/opengamma/strata/product/SecurityPriceInfo.java]: Should this be a `BigDecimal`?

File: modules/product/src/main/java/com/opengamma/strata/product/SecurityPriceInfo.java [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+  @PropertyDefinition(validate = "notNull")
+  private final CurrencyAmount tickValue;
+  /**
+   * The size of each contract.
+   * <p>
+   * Contract size is the quantity of the underlying present in each derivative contract.
+   * For example, an equity option typically consists of 100 shares.
+   */
+  @PropertyDefinition(validate = "ArgChecker.notNegativeOrZero")
+  private final double contractSize;
```
Comment (cjkent) [inline modules/product/src/main/java/com/opengamma/strata/product/SecurityPriceInfo.java]: `BigDecimal`?

File: modules/product/src/main/java/com/opengamma/strata/product/SecurityPriceInfo.java [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+   * For example, for USD this will set the tick size to 0.01 and the tick value to $0.01.
+   * This typically matches the conventions of equities and bonds.
+   * 
+   * @param currency  the currency to derive the price information from
+   * @return the security price information
+   */
+  public static SecurityPriceInfo ofCurrencyMinorUnit(Currency currency) {
+    int digits = currency.getMinorUnitDigits();
+    double unitAmount = 1;
+    for (int i = 0; i < digits; i++) {
```
Comment (cjkent) [inline modules/product/src/main/java/com/opengamma/strata/product/SecurityPriceInfo.java]: double unitAmount = Math.pow(10, -currency.getMinorUnitDigits()):

**Comment by cjkent** [comment]
That's everything

File: modules/product/src/main/java/com/opengamma/strata/product/SecurityPriceInfo.java (hunk 1 above)
Comment (jodastephen) [inline modules/product/src/main/java/com/opengamma/strata/product/SecurityPriceInfo.java]: No, in Strata we've accepted that everything is a `double`

**Pull request merged**